ext/js/pages/settings/text-transformations-controller.js

```diff
@@ -0,0 +1,171 @@
+        for (const transformation of settingsItems) {
+            transformation.remove();
+        }
+    }
+
+    /** */
+    _renderSettingsItems() {
+        for (const transformation of this._transformations) {
+            const settingsItem = this._createSettingsItem(transformation);
+            this._container.appendChild(settingsItem);
+        }
+    }
+
+    /**
+     * @param {import('language').TextTransformation} transformation
+     * @returns {HTMLElement}
+     */
+    _createSettingsItem(transformation) {
+        const settingsItem = document.createElement('div');
+        settingsItem.classList.add('settings-item', 'text-transformation');
+
+        const innerWrappableDiv = document.createElement('div');
+        innerWrappableDiv.classList.add('settings-item-inner', 'settings-item-inner-wrappable');
+
+        innerWrappableDiv.appendChild(this._createLeftSide(transformation));
+        innerWrappableDiv.appendChild(this._createRightSide(transformation));
+
+        settingsItem.appendChild(innerWrappableDiv);
+
+        return settingsItem;
+    }
+
+    /**
+     * @param {import('language').TextTransformation} transformation
+     * @returns {HTMLElement}
+     */
+    _createLeftSide(transformation) {
+        const leftSide = document.createElement('div');
+        leftSide.classList.add('settings-item-left');
+
+        leftSide.appendChild(this._createLabel(transformation));
+        leftSide.appendChild(this._createDescription(transformation));
+
+        return leftSide;
+    }
+
+    /**
+     * @param {import('language').TextTransformation} transformation
+     * @returns {HTMLElement}
+     */
+    _createLabel(transformation) {
+        const label = document.createElement('div');
+        label.classList.add('settings-item-label');
+        label.textContent = transformation.name;
+        return label;
+    }
+
+    /**
+     * @param {import('language').TextTransformation} transformation
+     * @returns {HTMLElement}
+     */
+    _createDescription(transformation) {
+        const description = document.createElement('div');
+        description.classList.add('settings-item-description');
+        description.textContent = transformation.description;
+        return description;
+    }
+
+    /**
+     * @param {import('language').TextTransformation} transformation
+     * @returns {HTMLElement}
+     */
+    _createRightSide(transformation) {
+        const rightSide = document.createElement('div');
+        rightSide.classList.add('settings-item-right');
+
+        rightSide.appendChild(this._createSelect(transformation));
+
+        return rightSide;
+    }
+
+    /**
+     * @param {import('language').TextTransformation} transformation
+     * @returns {HTMLSelectElement}
+     */
+    _createSelect(transformation) {
+        const select = document.createElement('select');
+        select.setAttribute('data-setting', `languages.${this._language}.textTransformations.${transformation.id}`);
```

Comment:
Not safe; escape string using `languages[${JSON.stringify(this._language)]...`

(Again this may also change if I recommend going forward with arrays for settings)